tuhi/gui/drawing.py

```diff
@@ -115,21 +124,33 @@ def _on_download_button_clicked(self, button):
         # Translators: filter to show svg files only
         filter_svg.set_name(_('SVG files'))
         filter_svg.add_pattern('*.svg')
+        filter_png = Gtk.FileFilter()
+        # Translators: filter to show svg files only
+        filter_png.set_name(_('PNG files'))
+        filter_png.add_pattern('*.png')
         dialog.add_filter(filter_svg)
+        dialog.add_filter(filter_png)
         dialog.add_filter(filter_any)
 
         response = dialog.run()
         if response == Gtk.ResponseType.OK:
             import shutil
 
-            # regenerate the SVG based on the current rotation.
-            # where we used the orientation buttons, we haven't updated the
-            # file itself.
-            self.process_svg()
-
             file = dialog.get_filename()
-            shutil.copyfile(self.svg.filename, file)
-            # FIXME: error handling
+
+            if file.lower().endswith('.png'):
+                # regenerate the PNG based on the current rotation.
+                # where we used the orientation buttons, we haven't updated the
+                # file itself.
+                self.process_png()
+                shutil.copyfile(self.png.filename, file)
```

Comment:
I'd argue that after copying the file we should delete it from our own data path. we use svg primarily in the GUI and that's why we keep those around, but png is a purely export feature so we can delete it after saving it.